cmd/handler-utils.go

```
@@ -116,16 +116,17 @@ func extractMetadataFromHeader(header http.Header) map[string]string {
 	}
 	// Go through all other headers for any additional headers that needs to be saved.
 	for key := range header {
-		cKey := http.CanonicalHeaderKey(key)
-		if strings.HasPrefix(cKey, "X-Amz-Meta-") {
-			metadata[cKey] = header.Get(key)
-		} else if strings.HasPrefix(key, "X-Minio-Meta-") {
-			metadata[cKey] = header.Get(key)
+		if key != http.CanonicalHeaderKey(key) {
```

I don't understand why this is needed - the only way to get a non-canonical header is if we add code that changes request headers. Request headers are not supposed to be modified - any code that does so is introducing bugs. So checking for this and returning errors to the user does not appear to be correct, as they could not have caused such a problem in the first place.

---

It is quite possible @donatello because this is a function and http.Header by nature is not protected. So anyone can modify and use it. This function is defining its minimum conditions not hoping that http.Header itself is never modified. 

IMO it is better to error out if we do indeed see an unexpected key since this function cannot operate in any other way.

---

Any thoughts @aead  further?

---

It's already addressed @deekoder look at the patch itself.

---

@y4m4 wanted to confirm with @aead. Please see Travis failed.